### PR TITLE
Sprint preperations for v2.4.x (DEV)

### PR DESCRIPTION
**Pull request by d4rken**
Preperations for the next sprint.
Groundwork, testing techspec assumptions and compatbility.

WIP

### CODE REVIEW

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptographyTest.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        val testData =
+            "P2mcDKDZmCbCp33tW+nBNx5SGirWi213ZwOqTUpyReDvRbNPaKnpirb2RBBj2xrbTGF8nG5uZcvi8tsNIkx7GaHvJEMKzXtokvLcpiC4PJTQiO6PDacXsg02+0V/NOmg69ewSsYfLlATv/bNmn7gK+HQpc4bhnWMPOLJSoASzQQZNtSCT7Y6V7gchKYlzZsknjE4u7b3hKh4NKdWMpCLG0H5I6QhYdKPdpMd7jMCmEX2dgRTxTRQc6Ne9XR32mDUWoZ2bHUIH/jaUvxNHVDrK9f2CM7Om1U/L4cKs0MiaU4L0YH1/5K1LPfgKMucUWvR7m4F5+e6+qlES+LDGmf2VNA/g0Rgt0S5cWYFk7067uVUtZYwUXb6qZn2jK6ycTunpTBGbXccH66wPeAUObo1FWHjJwU1Hw85WA9OjxLV4HYE1heLt2wgXna5HCqflz9TmKAlxK5tcOEPRIPIsvo+mXkY4yBTJqTB+YpawZvED77BxSlsrzI+1uRPfRf63CxA"
```
Comment (mtwalli) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptographyTest.kt:16]: Please break it down to multi lines text? detekt will complain about that in the  near future

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/encryption/rsa/RSAKeyTest.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
...M9QfGEP6hBS+pXJarYb6gXWGIoiVjl+heg66Gtccjk4LrTZhheIM/KBvSwgS0/qu61X7qWscS4eLYP+7JV4Fz1erNukujFd52qYnfoKIcKL1jbKkYad4anB5wPZCaTdkFLJx9eEbz5CM9KEfUYkigQPfSkNAoHBAJxGC7m3J8DGbtIXLLJcr4Pyk/NKrfSnQPWXj/5Plzr096Djm2HhAL1ty6StFU7+N0s8JrRoxovsvwYKB7XjaLiLeJNOwaXUZ0SHDB43DlJvLxHgWlxjJuSdOqHEDUpN8H0+GaBZoFBpxSAbusBAVf5NwajPttmLf0JUIKlpX86gEmHCClAdhKc5CAfRCv2f6EtKwDm6JXthUGEfF+m89bXi2rSsJTVJiwWq5i3JYnadqqvwnItij3IugahsxOLytQKBwD2BIS0+YqkEuFLpyUQXdZx3rzupp7nP2szs2Ij+b16c+Se7F0xTcSmECrAtYtU527PnFXec9FxYglRRVWeTlTxgn07oBynT+fgcE/RSqNO03q3GTnvfFc13qFj6E8rp7ngekUUf7UHQ9EDut0iv3mteU2JbUlFc9IVufWBUcaNRz0fSeouiLhqUYz1JPlP1S8IpS0O0Qo5O/SpA1OL3fX0J4IG3cB92UQaJuY4CqECh3TNLf94nBzGObl1DoIRuAQKBwAM/aWr4pXjaJt9y39jtGDCzz+NUf/z/pNf1yGZOnSP6h1LuqAIGvQ8ywvKvLiwwGFIV8+/35Xhu+SFVAAgFq915+I2HK3sYyPShodzW5BNOgDns4bPd06cYpWlLO8N5jy/rJHkyo6RILnKWYPEx5Red4hJyDqjCv2hPe0ZKDFWs+fTSJYi4tFMNWl9fNs1Cj39RFGGevryrrxH1pXeUF78p8cWjdjp/RsZdYN8+ui8g47UjEP3MlJKY6NTpPcZBqQKBwQDSEzYU8kjtAup5G+pzB9CvHzWf6t1SUa420VE9xoQb+l40U5KD+9Z3YRk/MFNoBFvM3/S0cTW+3hf1viUReXDC7Pomf8zDbaplVH6gwh3E3ajePBsQ/HmgRplwuxNo1K8YBsMfHYY+dirR+0cw2T/gObuY/95v+a8jezRgKXhgl5UctbXx7XM+89BsGGFQq5U0v91Flyq65LAO9SBaSHUqvaSRL9ASjpgH2tWGancbRfAqE0omu4b5x1zYVmY9d3I="
```
Comment (mtwalli) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/encryption/rsa/RSAKeyTest.kt:12]: Please break it down to multi lines text?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptography.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        privateKey: RSAKey.Private,
+        cipherType: CipherType = CipherType.RSA_PKCS1_OAEP_PADDING,
+    ): ByteString {
+
+        val cipher = Cipher.getInstance(cipherType.transformation).apply {
+            init(Cipher.DECRYPT_MODE, privateKey.speccedKey, cipherType.oaepParameterSpec)
+        }
+
+        return cipher.doFinal(toDecrypt.toByteArray()).toByteString().also {
+            Timber.v("Decrypted %s... bytes to %s... bytes", toDecrypt.size, it.size)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptography.kt:27]: why the dots after the count?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encoding/Base64Extensions.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package de.rki.coronawarnapp.util.encoding
+
+import okio.ByteString.Companion.toByteString
+
+fun ByteArray.base64() = toByteString().base64()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encoding/Base64Extensions.kt:5]: Small test please?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptography.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    interface CipherProvider {
+        fun createCipher(): Cipher
+    }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptography.kt:33]: Not used yet?

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptographyTest.kt (hunk 1 above)
Comment (d4rken) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptographyTest.kt:16]: Will use `@Suppress("MaxLineLength")` to keep the values  searchable when trying to compare.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptography.kt (hunk 3 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptography.kt:27]: Initially I wanted to log part of the key, this is a leftover, will fix.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptography.kt (hunk 5 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/encryption/rsa/RSACryptography.kt:33]: Leftover from previous approach :+1:

**Review by mtwalli — APPROVED** [review]
👍🏼

**Review by chiljamgossow — APPROVED** [review]
LGTM

**Pull request merged**